fastlane_core/lib/fastlane_core/configuration/config_item.rb

```diff
@@ -87,14 +87,12 @@ def valid?(value)
 
     # Returns an updated value type (if necessary)
     def auto_convert_value(value)
-      # Weird because of https://stackoverflow.com/questions/9537895/using-a-class-object-in-case-statement
-      case
-      when data_type == Array
+      if data_type == Integer
+        return value.to_i unless value.nil? || value.to_i.to_s != value.to_s
```

Comment:
Good point, let's stick with the existing solution then, but please refactor to make the order more clear 👍